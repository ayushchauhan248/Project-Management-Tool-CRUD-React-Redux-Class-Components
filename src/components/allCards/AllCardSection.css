.every{
  text-decoration: none;
  color: black;
}

#proDiv{
    display: flex;
    flex-direction: row; 
    background-color: rgb(214, 214, 214);
    padding: 20px;
    border-bottom: 3px solid white;
}

#cardButtonSections{
  padding:10px;
  display: flex;
  flex-direction: column;
  background: rgb(80, 73, 119);
  width: 150px;
  min-height: 100px;
  border-radius: 10px;
  border: 2px solid rgb(214, 214, 214)
}

.btndash{
  background-color: white;
  border-radius: 7px;
  border: 2px solid rgb(80, 73, 119);
  padding: 7px;
  cursor: pointer;
  font-size:medium;
  width: 100%;
}

.btndash:hover{
  background-color: rgb(80, 73, 119);
  border: 2px solid white ;
  color:white;
}

.linkClass {
  margin-top: 15px;
}

#head{
  font-size: xx-large;
  font-weight: 600;
  font-family: 'Times New Roman', Times, serif;
  padding:10px;
  text-align: center;
  color: rgb(80, 73, 119);
}

#allcard{
  z-index: -1;
  padding-left: 200px;
}

.create {
  position: absolute;
  top: 60px;
  right: 80px;
  overflow: hidden;
  position: fixed;
  cursor: pointer;
  border-radius: 100%;
}

.createClose{
  position: absolute;
  top:0px;
  right:0px;
}

#deleteNotifi{
  border: 1px solid rgb(80, 73, 119);
  background-color: lightblue;
  border-radius: 5px;
  padding: 5px;
  font-size: medium;
  color:rgb(80, 73, 119);
  margin-bottom: 10px;
}