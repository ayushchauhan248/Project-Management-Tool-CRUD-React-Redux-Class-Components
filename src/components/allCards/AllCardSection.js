import { Component } from "react";
import ProjectCard from "../projectCard/ProjectCard";
import "./AllCardSection.css";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import Fab from "@mui/material/Fab";
import AddIcon from "@mui/icons-material/Add";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../../theme";
import TextField from "@mui/material/TextField";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogTitle from "@mui/material/DialogTitle";
import Button from "@mui/material/Button";
import "../../pages/app/CreateProject.css";

class AllCardSection extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "",
      technology: "",
      deadline: "",
      description: "",
      createDailog: false,
    };
  }
  componentDidMount() {
    this.props.getProjectAction();
  }

  handleDelete = (id) => {
    this.props.deleteProjectAction(id);
    alert("Click OK to delete");
    window.location.reload();
  };
  handleClickOpen = () => {
    this.setState({ createDailog: true });
  };

  handleClose = () => {
    this.setState({ createDailog: false });
  };

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSubmit = (e) => {
    const project = {
      title: this.state.title,
      technology: this.state.technology,
      deadline: this.state.deadline,
      description: this.state.description,
    };
    e.preventDefault();
    this.props.setprojectAction(project);
    this.handleClose();
  };

  render() {
    const { title, technology, deadline, description } = this.state;
    const enabled =
      title.length > 0 &&
      technology.length > 0 &&
      deadline.length > 0 &&
      description.length > 0;
    console.log("this props all card", this.props.projects);
    return (
      <div id="allcard">
        <div id="head">ALL PROJECTS</div>
        <div className="create">
          <ThemeProvider theme={theme}>
            <Fab
              color="secondary"
              aria-label="add"
              onClick={this.handleClickOpen}
            >
              <AddIcon />
            </Fab>
          </ThemeProvider>
        </div>
        <Dialog open={this.state.createDailog} maxWidth="xl" maxHeight="xl">
          <div id="headi">+ CREATE PROJECT</div>
          <div id="formOuter">
            <form onSubmit={this.handleSubmit} id="formc">
              <div id="two">
                <input
                  type="text"
                  id="title"
                  name="title"
                  placeholder="Project Title"
                  className="title"
                  onChange={this.handleChange}
                  autoComplete="off"
                  ref={this.emailInputRef}
                />
                <input
                  type="text"
                  id="techno"
                  name="technology"
                  placeholder="Technology "
                  className="title"
                  autoComplete="off"
                  onChange={this.handleChange}
                />
              </div>
              <div id="three">
                <input
                  type="text"
                  id="deadline"
                  name="deadline"
                  placeholder="Deadline "
                  className="title"
                  autoComplete="off"
                  onChange={this.handleChange}
                />
                <textarea
                  type="text"
                  id="description"
                  name="description"
                  placeholder="Description "
                  className="title"
                  autoComplete="off"
                  onChange={this.handleChange}
                />
              </div>
              <div id="four">
                <button disabled={!enabled} type="submit" id="subBut">
                  Submit
                </button>
              </div>
            </form>
          </div>
        </Dialog>
        {this.props.projects.projectData?.reverse().map((res) => {
          return (
            <div id="proDiv">
              <ProjectCard
                title={res.title}
                technology={res.technology}
              ></ProjectCard>
              <div id="cardButtonSections">
                <Link to={`/project/${res._id}`} className="every linkClass">
                  <button className="btndash">View</button>
                </Link>
                <Link to={"/edit/" + res._id} className="every linkClass">
                  <button className="btndash">Edit</button>
                </Link>
                <button
                  onClick={() => this.handleDelete(res._id)}
                  className="btndash linkClass"
                  id="testing"
                >
                  Delete
                </button>
              </div>
            </div>
          );
        })}
      </div>
    );
  }
}

AllCardSection.propTypes = {
  projects: PropTypes.object.isRequired,
  getProjectAction: PropTypes.func.isRequired,
  deleteProjectAction: PropTypes.func.isRequired,
  setprojectAction: PropTypes.func.isRequired,
};

AllCardSection.defaultProps = {
  getProjectAction: () => {},
  deleteProjectAction: () => {},
  setprojectAction: () => {},
};

export default AllCardSection;
