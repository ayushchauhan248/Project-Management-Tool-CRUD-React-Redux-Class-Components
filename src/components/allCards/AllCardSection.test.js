import React from "react";
import AllCardSection from "./AllCardSection";
import { shallow, mount, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

describe("All Card Section tests", () => {
  it("snapshot testing", () => {
    const component = shallow(
      <AllCardSection
        projects={[
          {
            _id: "6246784888b2ac673d77f759",
            title: "TODO APP",
            technology: "Python flask",
            deadline: "1 may",
            description: "create ",
          },
          {
            _id: "6246786688b2ac673d77f75f",
            title: "Keep Notes",
            technology: "React Native",
            deadline: "15 April",
            description: "create a todo type app for keeping notes",
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("test checking ", () => {
    const component = shallow(
      <AllCardSection
        projects={[
          {
            _id: "6246784888b2ac673d77f759",
            title: "TODO APP",
            technology: "Python flask",
            deadline: "1 may",
            description: "create ",
          },
          {
            _id: "6246786688b2ac673d77f75f",
            title: "Keep Notes",
            technology: "React Native",
            deadline: "15 April",
            description: "create a todo type app for keeping notes",
          },
        ]}
      />
    );
    expect(component.find("#head").text()).toEqual("ALL PROJECTS");
  });
});
