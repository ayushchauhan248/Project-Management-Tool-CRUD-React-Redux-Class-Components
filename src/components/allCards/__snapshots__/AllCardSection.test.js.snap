// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Card Section tests snapshot testing 1`] = `
<div
  id="allcard"
>
  <div
    id="head"
  >
    ALL PROJECTS
  </div>
  <Link
    className="create"
    to="/create"
  >
    <BsPlusLg />
  </Link>
</div>
`;
